### PR TITLE
NFC: Code simplification

### PR DESCRIPTION
**Pull request by nouiz**
Simplify some XLA tests.

@cheshire

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/horizontal_loop_fusion_test.cc [hunk 1]
```diff
@@ -593,14 +589,10 @@ TEST_F(HorizontalLoopFusionTest, TraversalOrder) {
...
-       module->entry_computation()->instructions()) {
-    if (instr->opcode() == HloOpcode::kFusion) {
-      ++total_fusion_instrs;
-    }
-  }
-  EXPECT_EQ(total_fusion_instrs, 2);
+  auto is_fusion = [](const HloInstruction* instr) {
+                     return instr->opcode() == HloOpcode::kFusion;
+                   };
+  EXPECT_EQ(absl::c_count_if(module->entry_computation()->instructions(), is_fusion), 2);
```
Comment (cheshire) [inline tensorflow/compiler/xla/service/gpu/horizontal_loop_fusion_test.cc:595]: Why not inline the lambda? Or go even further and make this a helper reusable method of this test?

Comment (nouiz) [reply tensorflow/compiler/xla/service/gpu/horizontal_loop_fusion_test.cc]: done.

**Review by cheshire — APPROVED** [review]

**Comment by thomasjoerg** [comment]
A fix for the ARM CI build failure is underway.

**Comment by nouiz** [comment]
> A fix for the ARM CI build failure is underway.

Great. But up to now, this didn't prevented the merge. My PR was still being merged.
Fixing the ARM CI will save us many useless emails, thanks!

**Pull request merged**